skyportal/tests/api/test_sources.py

```diff
@@ -824,6 +824,103 @@ def test_source_photometry_summary_info(
     assert data["data"]["peak_detected_mag"] == mag1_ab
 
 
+def test_sources_include_detection_stats(
+    upload_data_token,
+    view_only_token,
+    public_source,
+    ztf_camera,
+    public_group,
+    upload_data_token_two_groups,
+    public_group2,
+):
+    # Some very high mjd to make this the latest point
+    # This is not a detection though
+    status, data = api(
+        'POST',
+        'photometry',
+        data={
+            'obj_id': str(public_source.id),
+            'mjd': 99999.0,
+            'instrument_id': ztf_camera.id,
+            'mag': None,
+            'magerr': None,
+            'limiting_mag': 22.3,
+            'magsys': 'ab',
+            'filter': 'ztfg',
+            'group_ids': [public_group.id],
+        },
+        token=upload_data_token,
+    )
+    assert status == 200
+    assert data['status'] == 'success'
+
+    # Another high mjd, but this time a photometry point not visible to the user
+    status, data = api(
+        'POST',
+        'photometry',
+        data={
+            'obj_id': str(public_source.id),
+            'mjd': 99900.0,
+            'instrument_id': ztf_camera.id,
+            'mag': None,
+            'magerr': None,
+            'limiting_mag': 22.3,
+            'magsys': 'ab',
+            'filter': 'ztfg',
+            'group_ids': [public_group2.id],
+        },
+        token=upload_data_token_two_groups,
+    )
+    assert status == 200
+    assert data['status'] == 'success'
+
+    # A high mjd, but lower than the first point
+    # Since this is a detection, it should be returned as "last_detected"
+    status, data = api(
+        'POST',
+        'photometry',
+        data={
+            'obj_id': str(public_source.id),
+            'mjd': 90000.0,
+            'instrument_id': ztf_camera.id,
+            'flux': 12.24,
+            'fluxerr': 0.031,
+            'zp': 25.0,
+            'magsys': 'ab',
+            'filter': 'ztfg',
+            'group_ids': [public_group.id],
+        },
+        token=upload_data_token,
+    )
+    assert status == 200
+    assert data['status'] == 'success'
+
+    status, data = api(
+        "GET",
+        "sources",
+        params={"includeDetectionStats": "true"},
+        token=view_only_token,
+    )
+    assert status == 200
+    assert data["status"] == "success"
+
+    assert any(
+        [
+            s["last_detected_at"] == arrow.get((90000.0 - 40_587) * 86400.0).isoformat()
+            for s in data["data"]["sources"]
+        ]
+    )
+    assert any(
+        [
+            s["peak_detected_at"] == arrow.get((90000.0 - 40_587) * 86400.0).isoformat()
+            for s in data["data"]["sources"]
+        ]
+    )
+
+    assert any([s["last_detected_mag"] is not None for s in data["data"]["sources"]])
```

Comment:
These could trigger if any of the sources posted during testing have any photometry points above threshold. 
Why not look for the specific magnitudes you put in?

Reply:
@guynir42 good call.